### PR TITLE
Update to allow for comments on home page

### PR DESCRIPTION
**Pull request by ElvisASC4**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,25 +27,39 @@
...
+
+        String json = convertToJson(comments);
+    
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+
+        response.sendRedirect("/index.html");
+    }
+
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (mmallett) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: You can make this easier to read with `Optional`

```java
private String getParameter(HttpServletRequest request, String name, String defaultValue) {
  return Optional.ofNullable(request.getParameter(name)).orElse(defaultValue);
}
```

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -50,7 +50,11 @@ <h1>Elvis Vasquez</h1>
         <div id="flower-image-container">
             <img class="" id="flower-image" src="">
         </div>
-        <button onclick="getHelloFromButton()">Click me if you're a cutie!</button>
+        <form action="/data" method="POST">
+            <p>Leave me a comment if you're brave >:):</p>
```
Comment (mmallett) [inline portfolio/src/main/webapp/index.html:54]: Nit: you can paste emojis into HTML >:)

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -64,9 +64,10 @@ function tryHarderAndGiveDaisy() {
 
 }
 
-async function getHelloFromButton() {
+async function getComments() {
     const response = await fetch('/data');
     const text = await response.json();
+    console.log(text);
```
Comment (mmallett) [inline portfolio/src/main/webapp/script.js:70]: No console.logs, use the debugger

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (ElvisASC4) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: This line of code broke my code. When I try to run the environment it gives me the error: cannot find symbol
[ERROR]   symbol:   variable Optional
[ERROR]   location: class com.google.sps.servlets.DataServlet

I assume my problem lies in how I define the global variable in order to use Optional. 
Public Static <T> Optional<T> ofNullable(T value); ?

Comment (mmallett) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: No you don't need to define any variable, just import the package, it's built into java

```java
import java.util.Optional

// some classes and things

Optional.ofNullable(request.getParameter(name)).orElse(defaultValue);
```

**Pull request merged**